core/templates/dev/head/components/forms/FormBuilderSpec.js

```diff
@@ -146,6 +169,212 @@ describe('Normalizer tests', function() {
   }));
 });
 
+describe('Testing requireIsFloat directive', function() {
+  var $compile, scope, testInput;
+
+  beforeEach(module('oppia'));
+
+  beforeEach(inject(function($compile, $rootScope) {
+    scope = $rootScope.$new();
+    var element = '<form name="testForm">' +
+      '<input name="floatValue" type="number" ng-model="localValue" ' +
+      'require-is-float apply-validation>' +
+      '</form>';
+    $compile(element)(scope);
+    testInput = scope.testForm.floatValue;
+  }));
+
+  it('should validate if value is a float', function() {
+    testInput.$setViewValue('2');
+    scope.$digest();
+    expect(testInput.$valid).toEqual(true);
+
+    testInput.$setViewValue('2.0');
+    scope.$digest();
+    expect(testInput.$valid).toEqual(true);
+
+    testInput.$setViewValue('3.5');
+    scope.$digest();
+    expect(testInput.$valid).toEqual(true);
+
+    testInput.$setViewValue('-3.5');
+    scope.$digest();
+    expect(testInput.$valid).toEqual(true);
+  });
+
+  it('should invalidate if value is not a float', function() {
+    testInput.$setViewValue('-abc');
+    scope.$digest();
+    expect(testInput.$valid).toEqual(false);
+
+    testInput.$setViewValue('3..5');
+    scope.$digest();
+    expect(testInput.$valid).toEqual(false);
+
+    testInput.$setViewValue('-2.abc');
+    scope.$digest();
+    expect(testInput.$valid).toEqual(false);
+
+    testInput.$setViewValue('0.3.5');
+    scope.$digest();
+    expect(testInput.$valid).toEqual(false);
+
+    testInput.$setViewValue(undefined);
+    scope.$digest();
+    expect(testInput.$valid).toEqual(false);
+  });
+});
+
+describe('Testing apply-validation directive', function() {
+  var $compile, element, scope, testInput;
+
+  beforeEach(module('oppia'));
+
+  beforeEach(inject(function($rootScope) {
+    scope = $rootScope.$new();
+    element = '<form name="testForm">' +
+      '<input name="inputValue" type="number" ng-model="localValue" ' +
+      'apply-validation>' +
+      '</form>';
+  }));
+
+  it('should apply isAtLeast validation', inject(function($compile) {
+    scope.validators = function() {
+      return [{
+        id: 'isAtLeast',
+        minValue: -2.5
+      }];
+    };
+    $compile(element)(scope);
+    testInput = scope.testForm.inputValue;
+
+    testInput.$setViewValue(-1);
+    scope.$digest();
+    expect(testInput.$valid).toEqual(true);
+
+    testInput.$setViewValue('1');
+    scope.$digest();
+    expect(testInput.$valid).toEqual(true);
+
+    testInput.$setViewValue(-3);
```

Comment:
In both this and the next test I think it makes sense to add a value that lies on the boundary (e.g. -2.5 in this case and 5 in the next)?